.github/workflows/joystream-node.yml

```diff
@@ -51,7 +50,10 @@ jobs:
   # Build and test joystream-node with all runtime profiles
   runtime_profiles:
     name: Runtime Profiles
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```

Comment:
It looks like `macos-latest` is still using `x86-64` VM (https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources), so I'm not sure if there's a difference.

Also it looks like GH doesn't offer any github-hosted ARM instances, we would need to use a self-hosted one.

Unless there's some easy solution I'm missing perhaps it's not worth tackling now then?

Reply:
Last time I checked macos-latest was infact using M1 processor, so arm64. I remember testing it in the create-release github flow and left a comment to that effect there: https://github.com/Joystream/joystream/blob/carthage/.github/workflows/create-release.yml#L18

And yes you are right, I didn't find a github-hosted linux arm64 instance, so for building the docker arm64 images we deploy AWS instance with arm64 arch `t4g` ec2 instance type: https://github.com/Joystream/joystream/blob/carthage/.github/workflows/joystream-node-docker.yml#L61

Reply:
I think the github docs might be out of date, I'm pretty sure binary created on macos-latest didn't work on my old mac laptop with x86_64, but worked fine on M1..

Reply:
I'll execute call-sizes in the flow to confirm.

Reply:
Following the roadmap I stumbled upon this post: https://github.blog/changelog/2022-08-09-github-actions-self-hosted-runners-now-support-apple-m1-hardware/, which explains that although the actions runner supports M1 and ARM64, GitHub doesn't provide a hosted runner, we'd need to set up our own self-hosted one:
> Now the [Actions runner](https://github.com/actions/runner) supports M1 and the ARM64 runtime meaning developers can run it on **their own M1 or M2 hardware**.

And:

> For additional information on how to set up a **self-hosted macOS ARM64 runner,** please refer to our [documentation](https://docs.github.com/en/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow#using-default-labels-to-route-jobs).

Reply:
I guess support still doesn't exist. https://github.com/actions/runner-images/issues/2187

Reply:
Ah thanks for looking into this. I suppose we can look into adding our own runner at a later date.